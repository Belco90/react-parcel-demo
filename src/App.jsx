import React, { useState, useEffect } from 'react'
import logo from './logo.svg'
import './App.css'

export const App = () => {
  const [secondsOpen, setSecondsOpen] = useState(0)
  const [clicksCount, setClicksCount] = useState(0)

  useEffect(() => {
    const timer = setInterval(
      () => setSecondsOpen((prevSecondsOpen) => prevSecondsOpen + 1),
      1000
    )
    return () => clearTimeout(timer)
  }, [])

  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="React logo" />
        <h1>Hello Parcel + React!</h1>
        <p>
          Edit <code>App.jsx</code> and save to test HMR updates.
        </p>
        <p>
          Page has been open for <code>{secondsOpen}</code> seconds.
        </p>
        <p>
          <button
            type="button"
            onClick={() => setClicksCount((count) => count + 1)}
          >
            Count is: {clicksCount}
          </button>
        </p>

        <p>
          <a
            className="App-link"
            href="https://beta.reactjs.org/"
            target="_blank"
            rel="noopener noreferrer"
          >
            Learn React
          </a>
          {' | '}
          <a
            className="App-link"
            href="https://parceljs.org/docs/"
            target="_blank"
            rel="noopener noreferrer"
          >
            Parcel Docs
          </a>
        </p>
      </header>
    </div>
  )
}
