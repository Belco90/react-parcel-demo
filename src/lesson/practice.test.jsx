/**
 * @fileoverview Figure out how to fix the linting errors reported in this file.
 */

import React from 'react'
import { render, screen, fireEvent } from '@testing-library/react'
import { App } from '../App'
import userEvent from '@testing-library/user-event'

it('should reset counts - using fireEvent', async () => {
  render(<App />)

  const countButton = screen.getByRole('button')
  expect(countButton).toHaveTextContent('Count is: 0')
  fireEvent.click(countButton)
  await fireEvent.click(countButton)
  expect(countButton).toHaveTextContent('Count is: 2')

  const resetInput = screen.getByLabelText(
    /write "reset" and hit enter to reset all the counts/i,
  )
  await fireEvent.keyDown(resetInput, {
    code: 'Enter',
    target: { value: 'something else' },
  })
  expect(countButton).toHaveTextContent('Count is: 2')

  await fireEvent.keyDown(resetInput, {
    target: { value: 'reset' },
  })
  expect(countButton).toHaveTextContent('Count is: 2')

  await fireEvent.keyDown(resetInput, {
    code: 'Enter',
  })
  expect(countButton).toHaveTextContent('Count is: 0')
})

it('should reset counts - using userEvent', async () => {
  render(<App />)

  const countButton = screen.getByRole('button')
  expect(countButton).toHaveTextContent('Count is: 0')
  userEvent.click(countButton)
  await userEvent.click(countButton)
  expect(countButton).toHaveTextContent('Count is: 2')

  const resetInput = screen.getByLabelText(
    /write "reset" and hit enter to reset all the counts/i,
  )
  await userEvent.type(resetInput, 're')
  await userEvent.type(resetInput, 'se', { delay: 200 })
  await userEvent.type(resetInput, 'tt', { delay: 0 })
  await userEvent.type(resetInput, '{backspace}', { delay: 15 })
  await userEvent.keyboard('{enter}')
  expect(countButton).toHaveTextContent('Count is: 0')
})
