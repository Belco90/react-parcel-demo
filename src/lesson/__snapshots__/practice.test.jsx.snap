// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should make a request to pokeapi 1`] = `
<div
  class="App"
>
  <header
    class="App-header"
  >
    <form>
      <label>
        Pick one Pokémon by their ID
        <div>
          <input
            max="898"
            min="1"
            name="pokemonId"
            required=""
            type="number"
          />
        </div>
        <button
          type="submit"
        >
          Search
        </button>
      </label>
    </form>
  </header>
</div>
`;

exports[`should make a request to pokeapi 2`] = `
<div
  class="App"
>
  <header
    class="App-header"
  >
    <form>
      <label>
        Pick one Pokémon by their ID
        <div>
          <input
            max="898"
            min="1"
            name="pokemonId"
            required=""
            type="number"
          />
        </div>
        <button
          type="submit"
        >
          Search
        </button>
      </label>
    </form>
  </header>
</div>
`;

exports[`should make a request to pokeapi 3`] = `
<div
  class="App"
>
  <header
    class="App-header"
  >
    <form>
      <label>
        Pick one Pokémon by their ID
        <div>
          <input
            max="898"
            min="1"
            name="pokemonId"
            required=""
            type="number"
          />
        </div>
        <button
          type="submit"
        >
          Search
        </button>
      </label>
    </form>
  </header>
</div>
`;

exports[`should make a request to pokeapi 4`] = `
<div
  class="App"
>
  <header
    class="App-header"
  >
    <form>
      <label>
        Pick one Pokémon by their ID
        <div>
          <input
            max="898"
            min="1"
            name="pokemonId"
            required=""
            type="number"
          />
        </div>
        <button
          type="submit"
        >
          Search
        </button>
      </label>
    </form>
  </header>
</div>
`;

exports[`should make a request to pokeapi 5`] = `
<div
  class="App"
>
  <header
    class="App-header"
  >
    <form>
      <label>
        Pick one Pokémon by their ID
        <div>
          <input
            max="898"
            min="1"
            name="pokemonId"
            required=""
            type="number"
          />
        </div>
        <button
          type="submit"
        >
          Search
        </button>
      </label>
    </form>
  </header>
</div>
`;

exports[`should make a request to pokeapi 6`] = `
<div
  class="App"
>
  <header
    class="App-header"
  >
    <form>
      <label>
        Pick one Pokémon by their ID
        <div>
          <input
            max="898"
            min="1"
            name="pokemonId"
            required=""
            type="number"
          />
        </div>
        <button
          type="submit"
        >
          Search
        </button>
      </label>
    </form>
  </header>
</div>
`;

exports[`should make a request to pokeapi 7`] = `
<div
  class="App"
>
  <header
    class="App-header"
  >
    <form>
      <label>
        Pick one Pokémon by their ID
        <div>
          <input
            max="898"
            min="1"
            name="pokemonId"
            required=""
            type="number"
          />
        </div>
        <button
          type="submit"
        >
          Search
        </button>
      </label>
    </form>
  </header>
</div>
`;

exports[`should make a request to pokeapi 8`] = `
<div
  class="App"
>
  <header
    class="App-header"
  >
    <form>
      <label>
        Pick one Pokémon by their ID
        <div>
          <input
            max="898"
            min="1"
            name="pokemonId"
            required=""
            type="number"
          />
        </div>
        <button
          type="submit"
        >
          Search
        </button>
      </label>
    </form>
  </header>
</div>
`;

exports[`should make a request to pokeapi 9`] = `
<div
  class="App"
>
  <header
    class="App-header"
  >
    <form>
      <label>
        Pick one Pokémon by their ID
        <div>
          <input
            max="898"
            min="1"
            name="pokemonId"
            required=""
            type="number"
          />
        </div>
        <button
          type="submit"
        >
          Search
        </button>
      </label>
    </form>
  </header>
</div>
`;

exports[`should make a request to pokeapi 10`] = `
<div
  class="App"
>
  <header
    class="App-header"
  >
    <form>
      <label>
        Pick one Pokémon by their ID
        <div>
          <input
            max="898"
            min="1"
            name="pokemonId"
            required=""
            type="number"
          />
        </div>
        <button
          type="submit"
        >
          Search
        </button>
      </label>
    </form>
  </header>
</div>
`;

exports[`should make a request to pokeapi 11`] = `
<div
  class="App"
>
  <header
    class="App-header"
  >
    <form>
      <label>
        Pick one Pokémon by their ID
        <div>
          <input
            max="898"
            min="1"
            name="pokemonId"
            required=""
            type="number"
          />
        </div>
        <button
          type="submit"
        >
          Search
        </button>
      </label>
    </form>
  </header>
</div>
`;

exports[`should make a request to pokeapi 12`] = `
<div
  class="App"
>
  <header
    class="App-header"
  >
    <form>
      <label>
        Pick one Pokémon by their ID
        <div>
          <input
            max="898"
            min="1"
            name="pokemonId"
            required=""
            type="number"
          />
        </div>
        <button
          type="submit"
        >
          Search
        </button>
      </label>
    </form>
  </header>
</div>
`;

exports[`should make a request to pokeapi 13`] = `
<div
  class="App"
>
  <header
    class="App-header"
  >
    <form>
      <label>
        Pick one Pokémon by their ID
        <div>
          <input
            max="898"
            min="1"
            name="pokemonId"
            required=""
            type="number"
          />
        </div>
        <button
          type="submit"
        >
          Search
        </button>
      </label>
    </form>
  </header>
</div>
`;

exports[`should make a request to pokeapi 14`] = `
<div
  class="App"
>
  <header
    class="App-header"
  >
    <form>
      <label>
        Pick one Pokémon by their ID
        <div>
          <input
            max="898"
            min="1"
            name="pokemonId"
            required=""
            type="number"
          />
        </div>
        <button
          type="submit"
        >
          Search
        </button>
      </label>
    </form>
  </header>
</div>
`;

exports[`should make a request to pokeapi 15`] = `
<div
  class="App"
>
  <header
    class="App-header"
  >
    <form>
      <label>
        Pick one Pokémon by their ID
        <div>
          <input
            max="898"
            min="1"
            name="pokemonId"
            required=""
            type="number"
          />
        </div>
        <button
          type="submit"
        >
          Search
        </button>
      </label>
    </form>
  </header>
</div>
`;

exports[`should make a request to pokeapi 16`] = `
<div
  class="App"
>
  <header
    class="App-header"
  >
    <form>
      <label>
        Pick one Pokémon by their ID
        <div>
          <input
            max="898"
            min="1"
            name="pokemonId"
            required=""
            type="number"
          />
        </div>
        <button
          type="submit"
        >
          Search
        </button>
      </label>
    </form>
  </header>
</div>
`;

exports[`should make a request to pokeapi 17`] = `
<div
  class="App"
>
  <header
    class="App-header"
  >
    <form>
      <label>
        Pick one Pokémon by their ID
        <div>
          <input
            max="898"
            min="1"
            name="pokemonId"
            required=""
            type="number"
          />
        </div>
        <button
          type="submit"
        >
          Search
        </button>
      </label>
    </form>
  </header>
</div>
`;

exports[`should make a request to pokeapi 18`] = `
<div
  class="App"
>
  <header
    class="App-header"
  >
    <form>
      <label>
        Pick one Pokémon by their ID
        <div>
          <input
            max="898"
            min="1"
            name="pokemonId"
            required=""
            type="number"
          />
        </div>
        <button
          type="submit"
        >
          Search
        </button>
      </label>
    </form>
  </header>
</div>
`;

exports[`should make a request to pokeapi 19`] = `
<div
  class="App"
>
  <header
    class="App-header"
  >
    <form>
      <label>
        Pick one Pokémon by their ID
        <div>
          <input
            max="898"
            min="1"
            name="pokemonId"
            required=""
            type="number"
          />
        </div>
        <button
          type="submit"
        >
          Search
        </button>
      </label>
    </form>
    <div>
      <img
        alt="Sprite of bulbasaur"
        src="https://pokemon.png"
      />
      <ul>
        <li>
          <strong>
            Name:
          </strong>
           
          bulbasaur
        </li>
        <li>
          <strong>
            Types:
          </strong>
           
          grass, poison
        </li>
      </ul>
    </div>
  </header>
</div>
`;
